hs/Dockerfile.haskell-build-artifacts

```diff
@@ -1,14 +1,50 @@
+FROM ethereum/solc:0.8.7-alpine  AS solc
```

Comment:
I would like all of the `0.8.7`s through this and the other files consolidated into one place that can be sourced by all the various Makefiles/etc (like `/DEPS` or something)